### DISCUSSION TITLE
Called when a panel has finished loading

### DISCUSSION THREAD

**Discussion by Benis67** [RomanKubiak/ctrlr#367] [General]
I need to set the value of some Modulators in the start-up phase. Usage Version = 6.0.26, for Windows
I used the following method, for a Test_INI panel (it is a simplified example to set ModTest = 16 on opening). I created a Function "called when a panel has finished loading" so done.

InitMetod = function (- [[CtrlrInstance -]] type)
panel: getModulatorByName ("ModTest"): getComponent (): setValue (16, true)
end

Everything is ok when I open the panel from ctrlr. Whatever was the last used value, ModTest is set to 16 after start-up.
It doesn't work if I create the windows executable .exe (file- export restricted instance) .
In this case the first time after execution the value is set correctly to 16; from the next time every time I run the .exe the last value that was before closing is remembered and it is not set to ModTest=16.
This value is read from the present Test_INI.settings file
c: \ Users \ XXXXXXXX \ AppData \ Roaming \ Test_INI \
Deleting that file obviously gets back to working.
How to solve the problem?
ps I also tried the various syntaxes indicated here but it does not work https://ctrlr.org/forums/topic/setmodulatorvalue-and-friends/
Here the panel and .exe www.benis.it/varie/Test_ini.zip

**Comment by dnaldoog** [comment]
Hi there @Benis67 

I don't know why that's happening and I haven't seen that before, but it was not working either when compiling an .exe file using version __5.3.201__. Incidentally I believe version 5.3.201 is best on Windows.

I believe I fixed it by delaying the assignment of the value _16_ to the _uiSlider_  on load by adding a timer to the panel. 

Seems to work now! 

If it doesn't on your version of Ctrlr try increasing the timer delay from my 100ms to a higher value. e.g. `  timer:startTimer(99,250)`

<pre>
InitMetod = function(--[[ CtrlrInstance --]] type)

    timer:setCallback (99,delayValueAssign)
    timer:startTimer(99,100)

end

function delayValueAssign(timerId)
    if timerId == 99 then 
	panel:getModulatorByName("ModTest"):getComponent():setValue(16, true)
        timer:stopTimer(timerId)
    end
end
</pre>
[Test_INI_1_0_beta01 DNALDOOG TIMER_2021-10-10_21-02.zip](https://github.com/RomanKubiak/ctrlr/files/7317372/Test_INI_1_0_beta01.DNALDOOG.TIMER_2021-10-10_21-02.zip)

**Reply by Benis67** [reply]
thanks a lot. I tried it and it also works with 6.0.26
I used as in the example
timer: startTimer (99,100)
